src/app/convai-checker.component.spec.ts

```diff
@@ -461,15 +461,24 @@ describe('Convai checker test', () => {
     let checker = fixture.componentInstance.checker;
     let queryText = 'Your mother was a hamster';
 
+    let mockScore = 0.3;
     let mockResponse: AnalyzeCommentResponse =
-      getMockCheckerResponse(checker.getToken(queryText));
+      getMockCheckerResponseWithScore(mockScore, checker.getToken(queryText));
 
     // Checks that the response is emitted.
     checker.analyzeCommentResponseChanged.subscribe(
       (emittedItem: AnalyzeCommentResponse|null) => {
         expect(emittedItem).toEqual(mockResponse);
     });
 
+    // Checks that the score change event is emitted.
+    // TODO(rachelrosen): Figure out a way to write these tests to guarantee
+    // that we wait for these subscribe calls; currently it seems async() and
+    // done() are not compatible.
```

Comment:
I'm not quite positive how applicable this is to your current code (I'm not quite familiar with the angular async for tests), but what I've seen before is doing the actual verification that a value has changed outside of the `.then` (well, `subscribe` in this case) and having the only code actually inside the `subscribe` be something like `mostRecentlyEmittedScore = emittedScore`.  Would something like that work?  Would also be great if you could walk me through what the asnyc on tests does...kind of hard to understand the motivation for it from the Angular docs.

Reply:
Yeah, I think that could possibly work, given that angular handles the async-ness of the emitting events before the whenStable() callback gets called. I'll give that a try!

Let's chat offline about the Angular async in tests, since it's pretty confusing and they're constantly changing things with the test framework so I believe the current internal state is different from the external one.

Reply:
Update. It worked, thanks!